### PR TITLE
Add `serde` support

### PR DESCRIPTION
**Pull request by gavrilikhin-d**
Need to add serde to `MietteDiagnostic` too, when #262 is merged

### CODE REVIEW

**Comment by gavrilikhin-d** [comment]
Implements #260

File: src/protocol.rs [hunk 1]
```diff
@@ -347,23 +419,22 @@ impl<'a> SpanContents<'a> for MietteSpanContents<'a> {
...
     offset: SourceOffset,
-    /// The total length of the span. Think of this as an offset from `start`.
-    length: SourceOffset,
+    /// The total length of the span
+    length: usize,
 }
 
 impl SourceSpan {
     /// Create a new [`SourceSpan`].
     pub fn new(start: SourceOffset, length: SourceOffset) -> Self {
```
Comment (gavrilikhin-d) [inline src/protocol.rs:434]: It should be:
```suggestion
    pub fn new(start: SourceOffset, length: usize) -> Self {
```
But will be breaking change. 

`SourceOffset` may not represent length in no way. This looks more like a range (e.g `3..3`)

Comment (zkat) [reply src/protocol.rs:434]: Can you make a pr for this once this lands? I'll mark it as breaking but that way we won't forget to do it. 

There's a good chunk of desirable breaking changes in the pipeline right now and I think I'm gonna find some time to roll up 6.0 at some point.

Comment (gavrilikhin-d) [reply src/protocol.rs:434]: Ok

**Comment by zkat** [comment]
Looks like you need to fix a conflict, but also I'm not sure about those other dependencies you bumped? Those seem to be what's causing the CI issues

**Comment by zkat** [comment]
Also you may as well add serde support to MietteDiagnostic while you're at it

**Comment by zkat** [comment]
Why do you need to update those dependencies?

**Comment by gavrilikhin-d** [comment]
I do't need to :) Just noticed they are outdated and thought it may easy to fix it

**Pull request merged**